docs/proposals/configuration_changes.md

```diff
@@ -0,0 +1,71 @@
+## Configuration Changes
+
+### Goal
+Create a package for creating and managing configuration values as a value store. This will be used to remove the necessity of each package to be loaded just for configuration values. The other benefit is that this package could be a more robust solution to the configuration, Allowing us to do a better job of watching for updates to the configmap and managing subscriptions to those changes. This value store should also be usable by other parts of the application and therefore will require less usage of hard coded configuration structures and will allow the addition or removal of configuration values to not create such an issue. 
+
+### Work Items
+* Create a specific package for managing configuration 
+* Create a package mutex map to be used to store configuration values.
+* Take structured yaml and create a map of map of map.... saving the key as the name of the config value
+```go
+m["broker"] = map["dev_broker": true...]
+m["registry"] = [map["name": "dh"...], map["name": "play"...]]
+```
+* Create a syntax, and document, for retrieving a single configuration value from the list. I.E `config.MustGetBoolConfig("broker.dev_broker")`
+* Create a public API that will give the caller multiple ways to retrieve data.
+    - `MustGet` -> returns a single value and will panic if not able to retrieve the data
```

Comment:
I'm not a fan of ```MustGet```.  I think a single API call makes more sense with the caller processing the possible return values and failures.